plenoptic/synthesize/geodesic.py

```diff
@@ -1,155 +1,340 @@
+        return fig
+
+    def populate_geodesic(self):
+        """Help format the current optimization variable `x` into a geodesic
+        attribute that can be used later.
+
+        It joins the endpoints, reshapes the variable into a sequence
+        of images, detaches the gradients and clips the range to [0, 1].
+        """
+        self.geodesic = torch.cat([self.xA, self.x, self.xB]
+                                  ).reshape(
+                                (self.n_steps+1, *self.image_shape[1:])
+                                            ).clamp(0, 1).detach()
+
+    def plot_deviation_from_line(self, video=None, figsize=(7, 5)):
+        """visual diagnostic of geodesic linearity in representation space.
+
+        Parameters
+        ----------
+        video : torch.Tensor, optional
+            natural video that bridges the anchor points
+        figsize : tuple, optional
+            set the dimension of the figure
+
+        Returns
+        -------
+        fig: matplotlib.figure.Figure
+        """
+        if not hasattr(self, 'geodesic'):
+            self.populate_geodesic()
+
+        fig, ax = plt.subplots(1, 1, figsize=figsize)
+        ax.plot(*deviation_from_line(self.model(self.pixelfade
+                                                ).view(self.n_steps+1, -1)
+                                     ), 'g-o', label='pixelfade')
+
+        deviation = deviation_from_line(
+                        self.model(self.geodesic
+                                   ).view(self.n_steps+1, -1))
+        ax.plot(*deviation, 'r-o', label='geodesic')
+
+        if video is not None:
+            ax.plot(*deviation_from_line(self.model(video
+                                                    ).view(self.n_steps+1, -1)
+                                         ), 'b-o', label='video')
+        ax.set(xlabel='distance along representation line',
+               ylabel='distance from representation line',
+               title='deviation from the straight line')
+        ax.legend(loc=1)
+
+        return fig
+
+    def _vector_jacobian_product(self, y, x, a):
+        """compute vector-jacobian product: $a^T dy/dx = dy/dx^T a$,
+        and allow for further gradient computations by retaining,
+        and creating the graph.
+        """
+        accJac = autograd.grad(y, x, a,
+                               retain_graph=True,
+                               create_graph=True)[0]
+        return accJac
+
+    def _step_jerkiness(self, y):
+        velocity = self._finite_difference(y)
+        acceleration = self._finite_difference(velocity)
+        acc_magnitude = torch.norm(acceleration, dim=1, keepdim=True)
+        acc_direction = torch.div(acceleration, acc_magnitude)
+        accJac = self._vector_jacobian_product(y[1:-1], self.x,
+                                               acc_direction)
+        step_jerkiness = torch.norm(accJac, dim=1) ** 2
+        return step_jerkiness
+
+    def calculate_jerkiness(self):
```

Comment:
does this need to be a public function? will a user ever call it?

Reply:
I will rewrite the docstring to make this clearer